brvs-core/src/main/java/iroha/validation/rest/RestService.java

```diff
@@ -98,41 +104,35 @@ public Response getStatus(@PathParam("txHash") String hash) {
     if (transactionVerdict == null) {
       transactionVerdict = ValidationResult.UNKNOWN;
     }
-    return Response.status(HttpStatus.SC_OK).entity(transactionVerdict).build();
+    return Response.ok(new ValidationResultResponse(transactionVerdict)).build();
   }
 
   @POST
   @Path("/register/{accountId}")
+  @Produces(MediaType.APPLICATION_JSON)
   public Response register(@PathParam("accountId") String accountId) {
     try {
       registrationProvider.register(accountId);
-    } catch (Exception e) {
-      return Response.status(HttpStatus.SC_UNPROCESSABLE_ENTITY).entity(e).build();
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
     }
-    return Response.status(HttpStatus.SC_NO_CONTENT).build();
+    return Response.ok(GenericStatusedResponse.SUCCESS).build();
   }
 
   @POST
   @Path("/isRegistered")
   @Consumes(MediaType.APPLICATION_JSON)
   @Produces(MediaType.APPLICATION_JSON)
   public Response isRegistered(String jsonBody) {
-    try {
-      final String accountId = ValidationUtils.gson
-          .fromJson(jsonBody, AccountIdJsonWrapper.class)
-          .getAccountId();
-      if (StringUtils.isEmpty(accountId)) {
-        throw new IllegalArgumentException("Invalid input");
-      }
-      final boolean isRegistered = registrationProvider.getRegisteredAccounts()
-          .stream()
-          .anyMatch(registeredAccount -> registeredAccount.equals(accountId));
-      return Response.status(HttpStatus.SC_OK)
-          .entity(ValidationUtils.gson.toJson(new AccountRegisteredBooleanWrapper(isRegistered)))
-          .build();
-    } catch (Exception e) {
-      return Response.status(HttpStatus.SC_UNPROCESSABLE_ENTITY).entity(e).build();
-    }
+    final String accountId = ValidationUtils.gson
+        .fromJson(jsonBody, AccountIdJsonWrapper.class)
+        .getAccountId();
+    fieldValidator.checkAccountId(accountId);
+    final boolean isRegistered = registrationProvider.getRegisteredAccounts()
+        .stream()
+        .anyMatch(registeredAccount -> registeredAccount.equals(accountId));
+    return Response.ok(ValidationUtils.gson.toJson(new AccountRegisteredResponse(isRegistered)))
```

Comment:
Why is it necessary to create JSON from an object using ValidationUtils in this place, and not directly, as in other places?

Reply:
Fixed